core/utils/checksum.h

```diff
@@ -74,6 +74,7 @@ static inline uint32_t CalculateSum(const void *buf, size_t len) {
   }
 #endif
 
+#if __x86_64
```

Comment:
Actually on second look, this code block is all x86 assembly code so it seems like x86_64 is the right thing to do, unless I misunderstand you.